images/linux/scripts/installers/java-tools.sh

```diff
@@ -8,50 +8,120 @@ source $HELPER_SCRIPTS/install.sh
 source $HELPER_SCRIPTS/os.sh
 source $HELPER_SCRIPTS/etc-environment.sh
 
-JAVA_VERSIONS_LIST=$(get_toolset_value '.java.versions | .[]')
-DEFAULT_JDK_VERSION=$(get_toolset_value '.java.default')
-JAVA_TOOLCACHE_PATH="$AGENT_TOOLSDIRECTORY/Java_Adopt_jdk"
-
-# Install GPG Key for Adopt Open JDK. See https://adoptopenjdk.net/installation.html
-wget -qO - "https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public" | apt-key add -
-add-apt-repository --yes https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/
-
-if isUbuntu18 ; then
-    # Install GPG Key for Azul Open JDK. See https://www.azul.com/downloads/azure-only/zulu/
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0xB1998361219BD9C9
-    apt-add-repository "deb https://repos.azul.com/azure-only/zulu/apt stable main"
-fi
+createEnvironmentVariable() {
+    local JAVA_VERSION=$1
+    local VENDOR_NAME=$2
+    local DEFAULT=$3
 
-apt-get update
+    case ${VENDOR_NAME} in
 
-for JAVA_VERSION in ${JAVA_VERSIONS_LIST[@]}; do
-    apt-get -y install adoptopenjdk-$JAVA_VERSION-hotspot=\*
-    javaVersionPath="/usr/lib/jvm/adoptopenjdk-${JAVA_VERSION}-hotspot-amd64"
-    echo "JAVA_HOME_${JAVA_VERSION}_X64=$javaVersionPath" | tee -a /etc/environment
-    fullJavaVersion=$(cat "$javaVersionPath/release" | grep "^SEMANTIC" | cut -d "=" -f 2 | tr -d "\"" | tr "+" "-")
+        "Adopt" )
+            INSTALL_PATH_PATTERN="/usr/lib/jvm/adoptopenjdk-${JAVA_VERSION}-hotspot-amd64" ;;
 
-    # If there is no semver in java release, then extract java version from -fullversion
-    if [[ -z $fullJavaVersion ]]; then
-        fullJavaVersion=$(java -fullversion 2>&1 | tr -d "\"" | tr "+" "-" | awk '{print $4}')
+        "Temurin-Hotspot" )
+            INSTALL_PATH_PATTERN="/usr/lib/jvm/temurin-${JAVA_VERSION}-jdk-amd64" ;;
+        *)
+            echo "Unknown vendor"
+            exit 1
+
+    esac
+
+    if [[ ${DEFAULT} == "True" ]]; then
+        echo "Setting up JAVA_HOME variable to ${INSTALL_PATH_PATTERN}"
+        echo "JAVA_HOME=${INSTALL_PATH_PATTERN}" | tee -a /etc/environment
```

Comment:
we have a function for that in helpers https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/helpers/etc-environment.sh#L17